controllers/volumegroup_controller.go

```diff
@@ -150,17 +150,25 @@ func (r *VolumeGroupReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
-	if err = r.removeVolumesFromVG(logger, instance); err != nil {
-		return ctrl.Result{}, utils.HandleErrorMessage(logger, r.Client, instance, err, removingPVC)
-	}
-	if err = r.addMatchingVolumesToVG(logger, instance); err != nil {
-		return ctrl.Result{}, utils.HandleErrorMessage(logger, r.Client, instance, err, addingPVC)
+	err = r.updatePVCs(err, logger, instance)
+	if err != nil {
+		return ctrl.Result{}, err
```

Comment:
we need to use `utils.HandleErrorMessage` function to add an event in case of an error (can be done in a separate ticket because of GA)

Reply:
CSI-5229